modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/workflow-definitions/evp-payment-confirmation/workflow-definition.xml

```diff
@@ -0,0 +1,123 @@
+<?xml version="1.0"?>
+
+<workflow-definition
+	xmlns="urn:liferay.com:liferay-workflow_7.4.0"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="urn:liferay.com:liferay-workflow_7.4.0 http://www.liferay.com/dtd/liferay-workflow-definition_7_4_0.xsd"
+>
+	<version>1</version>
+	<state>
+		<name>7cc1ce8f-188a-4b03-9610-b8fa64755e3c</name>
+		<description>Begin a workflow.</description>
+		<metadata>
+			<![CDATA[
+				{
+					"xy": [
+						48.00006103515625,
+						125.00000762939453
+					]
+				}
+			]]>
+		</metadata>
+		<actions>
+			<action>
+				<name>Pay</name>
+				<description></description>
+				<script>
+					<![CDATA[import com.liferay.object.model.ObjectDefinition;
+import com.liferay.object.service.ObjectDefinitionLocalServiceUtil;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+import com.liferay.portal.kernel.workflow.WorkflowStatusManagerUtil;
+import com.liferay.portal.kernel.service.UserLocalServiceUtil;
+import com.liferay.object.model.ObjectEntry;
+import com.liferay.object.service.ObjectEntryLocalServiceUtil;
+import org.json.JSONObject;
+import java.io.Serializable;
+import java.util.Map;
+
+long classPK = GetterUtil.getLong((String) workflowContext.get(
+WorkflowConstants.CONTEXT_ENTRY_CLASS_PK));
+
+ObjectEntry currentyObjectEntry = ObjectEntryLocalServiceUtil.getObjectEntry(classPK);
+
+Map < String, Serializable > currentObjectValues = currentyObjectEntry.getValues();
+
+JSONObject jsonObject = new JSONObject(currentObjectValues);
+
+long requestId = jsonObject.getLong("r_requestId_c_evpRequestId");
```

Comment:
why you need to convert to json? you can simply use as a map and get the id and cast to long

Reply:
Done!